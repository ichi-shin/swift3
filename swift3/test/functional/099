#!/bin/bash
# Copyright(c)2014 NTT corp. All Rights Reserved.
# DB1-1-2-2_3

. ./common

S3USER=tester

_s3_put /bucket -H "x-amz-grant-full-control: id='${TENANT}:${TESTER_USER},id='${TENANT}:${TESTER2_USER}',id='${TENANT2}:${TESTER3_USER}'"
_s3_get /bucket -D - | _filter_curl xml

S3USER=tester2

echo data > $tmp.data
MD5=$(_md5 $tmp.data) CONTENT_TYPE='application/octet-stream' \
    _s3_put /bucket/objectA -T $tmp.data \
    -H 'Content-Disposition: attachment; filename="a.txt.gz"' \
    -H 'Content-Encoding: gzip' \
    -H "Content-Length: $(_file_size $tmp.data)" \
    -H 'Expect: 100-continue' \
    -H 'x-amz-storage-class: STANDARD' \
    -H "x-amz-grant-full-control: id='${TENANT2}:${TESTER3_USER}" -D - | _filter_curl
_s3_get /bucket/objectA?acl -D - | _filter_curl xml

S3USER=tester3

_s3_put /bucket/objectA?acl -H "x-amz-grant-read: uri='http://acs.amazonaws.com/groups/global/AllUsers'"

S3USER=tester2

_s3_get /bucket/objectA?acl -D - | _filter_curl xml
