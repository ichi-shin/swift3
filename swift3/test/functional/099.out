QA output created by 099
> s3curl --id tester -- -X PUT -H x-amz-grant-full-control: id='TESTER,id='TESTER2',id='TESTER3' http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X GET -D - http://SWIFT_HOST/bucket... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: application/xml
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<ListBucketResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Name>bucket</Name>
  <Prefix/>
  <Marker/>
  <MaxKeys>1000</MaxKeys>
  <IsTruncated>false</IsTruncated>
</ListBucketResult>
> s3curl --id tester2 --contentMd5 YTfN5Ik8WfdvAFqBI9jo5g== --contentType application/octet-stream -- -X PUT -T /TMP.data -H Content-Disposition: attachment; filename="a.txt.gz" -H Content-Encoding: gzip -H Content-Length: 5 -H Expect: 100-continue -H x-amz-storage-class: STANDARD -H x-amz-grant-full-control: id='TESTER3 -D - http://SWIFT_HOST/bucket/objectA... 200
HTTP/1.1 100 Continue

HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
ETag: "ETAG"
Last-Modified: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

> s3curl --id tester2 -- -X GET -D - http://SWIFT_HOST/bucket/objectA?acl... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<AccessControlPolicy xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Owner>
    <ID>TESTER2</ID>
    <DisplayName>TESTER2</DisplayName>
  </Owner>
  <AccessControlList>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="CanonicalUser">
        <ID>TESTER3</ID>
        <DisplayName>TESTER3</DisplayName>
      </Grantee>
      <Permission>FULL_CONTROL</Permission>
    </Grant>
  </AccessControlList>
</AccessControlPolicy>
> s3curl --id tester3 -- -X PUT -H x-amz-grant-read: uri='http://acs.amazonaws.com/groups/global/AllUsers' http://SWIFT_HOST/bucket/objectA?acl... 200
> s3curl --id tester2 -- -X GET -D - http://SWIFT_HOST/bucket/objectA?acl... 200
HTTP/1.1 200 OK
Content-Length: LENGTH
Content-Type: text/html; charset=UTF-8
Date: DATE
x-amz-id-2: TXID
x-amz-request-id: TXID
X-Trans-Id: TXID

<?xml version='1.0' encoding='UTF-8'?>
<AccessControlPolicy xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <Owner>
    <ID>TESTER2</ID>
    <DisplayName>TESTER2</DisplayName>
  </Owner>
  <AccessControlList>
    <Grant>
      <Grantee xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="Group">
        <URI>http://acs.amazonaws.com/groups/global/AllUsers</URI>
      </Grantee>
      <Permission>READ</Permission>
    </Grant>
  </AccessControlList>
</AccessControlPolicy>
